### PR TITLE
[Sprint 1] Add Cream Decoration in Cake

### PR DESCRIPTION
**Pull request by Tuanlvt**
Add Cream Decoration in Cake

### CODE REVIEW

File: app/src/main/java/com/fstyle/cakezy/screen/decorationscake/CreamAdapter.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        }
+        mICreamCakeBitmaps.addAll(creamCakeBitmaps);
+        notifyDataSetChanged();
+    }
+
+    void setItemClickListener(OnRecyclerViewItemClickListener<ICreamCakeBitmap> itemClickListener) {
+        mItemClickListener = itemClickListener;
+    }
+
+    static class ItemViewHolder extends RecyclerView.ViewHolder {
```
Comment (trainer-android-fr) [inline app/src/main/java/com/fstyle/cakezy/screen/decorationscake/CreamAdapter.java]: WARNING: Missing a Javadoc comment.

File: app/src/main/java/com/fstyle/cakezy/screen/decorationscake/adapterCandle/CandleAdapter.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
+package com.fstyle.cakezy.screen.decorationscake.adapterCandle;
```
Comment (trainer-android-fr) [inline app/src/main/java/com/fstyle/cakezy/screen/decorationscake/adapterCandle/CandleAdapter.java]: ERROR: Name 'com.fstyle.cakezy.screen.decorationscake.adapterCandle' must match pattern '^[a-z]+(\.[a-z][a-z0-9_]{1,})*$'.

File: app/src/main/java/com/fstyle/cakezy/screen/decorationscake/adapterCandle/CandleAdapter.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        }
+        mCandleCakes.addAll(candleCakes);
+        notifyDataSetChanged();
+    }
+
+    public void setItemClickListener(OnRecyclerViewItemClickListener<Object> itemClickListener) {
+        mItemClickListener = itemClickListener;
+    }
+
+    static class ItemViewHolder extends RecyclerView.ViewHolder {
```
Comment (trainer-android-fr) [inline app/src/main/java/com/fstyle/cakezy/screen/decorationscake/adapterCandle/CandleAdapter.java]: WARNING: Missing a Javadoc comment.

File: app/src/main/java/com/fstyle/cakezy/screen/decorationscake/adapterCandle/ItemCandleViewModel.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
+package com.fstyle.cakezy.screen.decorationscake.adapterCandle;
```
Comment (trainer-android-fr) [inline app/src/main/java/com/fstyle/cakezy/screen/decorationscake/adapterCandle/ItemCandleViewModel.java]: ERROR: Name 'com.fstyle.cakezy.screen.decorationscake.adapterCandle' must match pattern '^[a-z]+(\.[a-z][a-z0-9_]{1,})*$'.

File: app/src/main/java/com/fstyle/cakezy/screen/decorationscake/adapterCream/CreamAdapter.java [hunk 5]
```diff
@@ -0,0 +1,80 @@
+package com.fstyle.cakezy.screen.decorationscake.adapterCream;
```
Comment (trainer-android-fr) [inline app/src/main/java/com/fstyle/cakezy/screen/decorationscake/adapterCream/CreamAdapter.java]: ERROR: Name 'com.fstyle.cakezy.screen.decorationscake.adapterCream' must match pattern '^[a-z]+(\.[a-z][a-z0-9_]{1,})*$'.

File: app/src/main/java/com/fstyle/cakezy/screen/decorationscake/adapterCream/CreamAdapter.java [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+        }
+        mCreamCakes.addAll(creamCakeBitmaps);
+        notifyDataSetChanged();
+    }
+
+    public void setItemClickListener(OnRecyclerViewItemClickListener<Object> itemClickListener) {
+        mItemClickListener = itemClickListener;
+    }
+
+    static class ItemViewHolder extends RecyclerView.ViewHolder {
```
Comment (trainer-android-fr) [inline app/src/main/java/com/fstyle/cakezy/screen/decorationscake/adapterCream/CreamAdapter.java]: WARNING: Missing a Javadoc comment.

File: app/src/main/java/com/fstyle/cakezy/screen/decorationscake/adapterCream/ItemDecorationViewModel.java [hunk 7]
```diff
@@ -0,0 +1,38 @@
+package com.fstyle.cakezy.screen.decorationscake.adapterCream;
```
Comment (trainer-android-fr) [inline app/src/main/java/com/fstyle/cakezy/screen/decorationscake/adapterCream/ItemDecorationViewModel.java]: ERROR: Name 'com.fstyle.cakezy.screen.decorationscake.adapterCream' must match pattern '^[a-z]+(\.[a-z][a-z0-9_]{1,})*$'.

File: app/src/main/java/com/fstyle/cakezy/screen/decorationscake/adapterCandle/CandleAdapter.java (hunk 2 above)
Comment (trainer-android-fr) [inline app/src/main/java/com/fstyle/cakezy/screen/decorationscake/adapterCandle/CandleAdapter.java]: WARNING: [PackageCase] Package name contains upper case characters
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/naming.html#PackageCase

File: app/src/main/java/com/fstyle/cakezy/screen/decorationscake/adapterCandle/ItemCandleViewModel.java (hunk 4 above)
Comment (trainer-android-fr) [inline app/src/main/java/com/fstyle/cakezy/screen/decorationscake/adapterCandle/ItemCandleViewModel.java]: WARNING: [PackageCase] Package name contains upper case characters
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/naming.html#PackageCase

File: app/src/main/java/com/fstyle/cakezy/screen/decorationscake/adapterCream/CreamAdapter.java (hunk 5 above)
Comment (trainer-android-fr) [inline app/src/main/java/com/fstyle/cakezy/screen/decorationscake/adapterCream/CreamAdapter.java]: WARNING: [PackageCase] Package name contains upper case characters
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/naming.html#PackageCase

File: app/src/main/java/com/fstyle/cakezy/screen/decorationscake/adapterCream/ItemDecorationViewModel.java (hunk 7 above)
Comment (trainer-android-fr) [inline app/src/main/java/com/fstyle/cakezy/screen/decorationscake/adapterCream/ItemDecorationViewModel.java]: WARNING: [PackageCase] Package name contains upper case characters
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/naming.html#PackageCase

File: app/src/main/res/values/colors.xml [hunk 8]
```diff
@@ -10,4 +10,5 @@
     <color name="color_red_white_light">#FFEBEE</color>
     <color name="color_red_dark">#B71C1C</color>
     <color name="color_white_translate">#59FFFFFF</color>
+    <color name="color_gray_translate">#59616161</color>
```
Comment (trainer-android-fr) [inline app/src/main/res/values/colors.xml]: WARNING: [UnusedResources] The resource `R.color.color_gray_translate` appears to be unused
 Unused resources make applications larger and slow down builds.

File: app/src/main/res/values/dimens.xml [hunk 9]
```diff
@@ -10,4 +10,6 @@
     <dimen name="dp_1">1dp</dimen>
     <dimen name="dp_10">10dp</dimen>
     <dimen name="dp_2">2dp</dimen>
+    <dimen name="dp_80">80dp</dimen>
+    <dimen name="dp_30">30dp</dimen>
```
Comment (trainer-android-fr) [inline app/src/main/res/values/dimens.xml]: WARNING: [UnusedResources] The resource `R.dimen.dp_30` appears to be unused
 Unused resources make applications larger and slow down builds.

File: app/src/main/java/com/fstyle/cakezy/screen/decorationscake/adaptercandle/CandleAdapter.java [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+        }
+        mCandleCakes.addAll(candleCakes);
+        notifyDataSetChanged();
+    }
+
+    public void setItemClickListener(OnRecyclerViewItemClickListener<Object> itemClickListener) {
+        mItemClickListener = itemClickListener;
+    }
+
+    static class ItemViewHolder extends RecyclerView.ViewHolder {
```
Comment (trainer-android-fr) [inline app/src/main/java/com/fstyle/cakezy/screen/decorationscake/adaptercandle/CandleAdapter.java]: WARNING: Missing a Javadoc comment.

File: app/src/main/java/com/fstyle/cakezy/screen/decorationscake/adaptercream/CreamAdapter.java [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+        }
+        mCreamCakes.addAll(creamCakeBitmaps);
+        notifyDataSetChanged();
+    }
+
+    public void setItemClickListener(OnRecyclerViewItemClickListener<Object> itemClickListener) {
+        mItemClickListener = itemClickListener;
+    }
+
+    static class ItemViewHolder extends RecyclerView.ViewHolder {
```
Comment (trainer-android-fr) [inline app/src/main/java/com/fstyle/cakezy/screen/decorationscake/adaptercream/CreamAdapter.java]: WARNING: Missing a Javadoc comment.

File: app/src/main/java/com/fstyle/cakezy/screen/decorationscake/DecorationsCakeActivity.java [hunk 12]
```diff
@@ -18,6 +19,7 @@
 
     @Inject
     DecorationsCakeContract.ViewModel mViewModel;
+    DesignCakeView mDesignCakeView;
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/cakezy/screen/decorationscake/DecorationsCakeActivity.java]: remove

File: app/src/main/java/com/fstyle/cakezy/screen/decorationscake/DecorationsCakeViewModel.java [hunk 13]
```diff
@@ -22,4 +54,43 @@ public void onStart() {
...
+        return mCreamAdapter;
+    }
+
+    @Override
+    public CandleAdapter getCandleAdapter() {
+        return mCandleAdapter;
+    }
+
+    private void init() {
+        List<CreamCake> creamCake = new ArrayList<>();
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/cakezy/screen/decorationscake/DecorationsCakeViewModel.java]: pls add TODO in here!

File: app/src/main/java/com/fstyle/cakezy/screen/decorationscake/adaptercandle/CandleAdapter.java [hunk 14]
```diff
@@ -0,0 +1,80 @@
...
+                OnRecyclerViewItemClickListener<Object> listener) {
+            super(binding.getRoot());
+            mContext = context;
+            mBinding = binding;
+            mItemClickListener = listener;
+        }
+
+        void bind(CandleCake candleCake) {
+            mBinding.setViewModel(
+                    new ItemCandleViewModel(mContext, candleCake, mItemClickListener));
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/cakezy/screen/decorationscake/adaptercandle/CandleAdapter.java]: remove Context

File: app/src/main/java/com/fstyle/cakezy/screen/choosestylecake/ChooseStyleCakeContract.java [hunk 15]
```diff
@@ -6,11 +6,15 @@
...
  * This specifies the contract between the view and the presenter.
  */
-interface ChooseStyleCakeContract {
+public interface ChooseStyleCakeContract {
     /**
      * View.
      */
     interface ViewModel extends BaseViewModel {
+        //XML
+        void onClickNext();
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/cakezy/screen/choosestylecake/ChooseStyleCakeContract.java]: Change to ->``onNextButtonClick()``

File: app/src/main/java/com/fstyle/cakezy/screen/decorationscake/DecorationsCakeModule.java [hunk 16]
```diff
@@ -22,13 +25,34 @@ public DecorationsCakeModule(@NonNull Activity activity) {
     @ActivityScope
     @Provides
     public DecorationsCakeContract.ViewModel provideViewModel(
-            DecorationsCakeContract.Presenter presenter) {
-        return new DecorationsCakeViewModel(presenter);
+
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/cakezy/screen/decorationscake/DecorationsCakeModule.java]: remove blank line

File: app/src/main/java/com/fstyle/cakezy/screen/decorationscake/adaptercandle/CandleAdapter.java [hunk 17]
```diff
@@ -0,0 +1,78 @@
...
+import com.fstyle.cakezy.screen.BaseRecyclerViewAdapter;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by Tuanlvt on 08/11/2017.
+ */
+
+public class CandleAdapter extends BaseRecyclerViewAdapter<CandleAdapter.ItemViewHolder> {
+    private final List<CandleCake> mCandleCakes;
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/cakezy/screen/decorationscake/adaptercandle/CandleAdapter.java]: init values for this variable in here!

File: app/src/main/java/com/fstyle/cakezy/screen/decorationscake/adaptercream/CreamAdapter.java [hunk 18]
```diff
@@ -0,0 +1,78 @@
...
+import com.fstyle.cakezy.screen.BaseRecyclerViewAdapter;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by Tuanlvt on 08/11/2017.
+ */
+
+public class CreamAdapter extends BaseRecyclerViewAdapter<CreamAdapter.ItemViewHolder> {
+    private final List<CreamCake> mCreamCakes;
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/cakezy/screen/decorationscake/adaptercream/CreamAdapter.java]: like above

File: app/src/main/java/com/fstyle/cakezy/screen/decorationscake/adaptercream/CreamAdapter.java [hunk 19]
```diff
@@ -0,0 +1,78 @@
...
+    private OnRecyclerViewItemClickListener<Object> mItemClickListener;
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/cakezy/screen/decorationscake/adaptercream/CreamAdapter.java]: @Setter above this line

File: app/src/main/java/com/fstyle/cakezy/screen/decorationscake/adaptercream/CreamAdapter.java [hunk 20]
```diff
@@ -0,0 +1,78 @@
...
+
+    public void updateData(List<CreamCake> creamCakeBitmaps) {
+        if (creamCakeBitmaps == null) {
+            return;
+        }
+        mCreamCakes.addAll(creamCakeBitmaps);
+        notifyDataSetChanged();
+    }
+
+    public void setItemClickListener(OnRecyclerViewItemClickListener<Object> itemClickListener) {
```
Comment (daolq3012) [inline app/src/main/java/com/fstyle/cakezy/screen/decorationscake/adaptercream/CreamAdapter.java]: remove this method

File: app/build.gradle [hunk 21]
```diff
@@ -81,7 +81,9 @@ dependencies {
   implementation dependencies.loggingInterceptor
   implementation dependencies.cardView
   implementation dependencies.glide
+  implementation dependencies.lombok
 
+  annotationProcessor "org.projectlombok:lombok:1.16.18"
```
Comment (daolq3012) [inline app/build.gradle]: pls use annotationProcessor dependencies.lombok and remove blank line (85)

**Pull request merged**